eth/downloader/beaconsync.go

```diff
@@ -266,8 +278,18 @@ func (d *Downloader) fetchBeaconHeaders(from uint64) error {
 		)
 		for i := 0; i < maxHeadersProcess && from <= head.Number.Uint64(); i++ {
 			header := d.skeleton.Header(from)
+
+			// The header is not found in skeleton space, try to find it in local chain.
+			if header == nil && from < tail.Number.Uint64() {
+				dist := tail.Number.Uint64() - from
+				if len(localHeaders) >= int(dist) {
+					header = localHeaders[dist-1]
```

Comment:
In theory, we should check the number of retrieved pivot header. But I think it should always be correct unless we have the programming fault.